concepts/conceptpower.py

```diff
@@ -0,0 +1,87 @@
+import requests,json
+import xml.etree.ElementTree as ET
+from requests.auth import HTTPBasicAuth
+
+
+class Conceptpower:
+    # Default behavior is to leave these to the constructor.
+    endpoint = None
+    namespace = None
+
+    def __init__(self, **kwargs):
+        # Give first priority to the class definition, if endpoint or namespace are defined (and not None).
+        if self.endpoint is None:
+            self.endpoint = kwargs.get(
+                "endpoint", "http://chps.asu.edu/conceptpower/rest/")
+
+        if self.namespace is None:
+            self.namespace = kwargs.get(
+               "namespace", "{http://www.digitalhps.org/}")
+
+    def search(self, params=None, headers=None):
+        url = "{0}ConceptSearch".format(self.endpoint)
+
+        response = requests.get(url, headers=headers, params=params)
+        concepts = []
+        if response.status_code == 200:
```

Comment:
if not, we definitely want to log what else is returned and an error message if there is one.

Reply:
also, you want to use a constant (`requests` probably has one) and not a magic number (even though it's an obvious one if you have done web application development before).